gammapy/cube/tests/test_models.py

```diff
@@ -37,14 +50,20 @@ def background(geom):
 
 
 @pytest.fixture(scope='session')
-def sky_model():
-    spatial_model = SkyGaussian(
-        lon_0='3 deg', lat_0='4 deg', sigma='3 deg',
-    )
-    spectral_model = PowerLaw(
-        index=2, amplitude='1e-11 cm-2 s-1 TeV-1', reference='1 TeV',
-    )
-    return SkyModel(spatial_model, spectral_model)
+def edisp(geom):
+    e_true = geom.get_axis_by_name('energy').edges
+    return EnergyDispersion.from_diagonal_matrix(e_true=e_true)
+
+
+@pytest.fixture(scope='session')
+def psf(geom):
+    sigma = 0.5 * u.deg
+    return PSFKernel.from_gauss(geom, sigma)
+
+
+@pytest.fixture()
```

Comment:
I've seen this new scheme of setting up tests in several places in gammapy. I don't think it's a good idea to put the class that is beeing tested in a fixture. The purpose of test fixtures is to provide a fixed baseline upon which tests can reliably and repeatedly execute ([source](https://docs.pytest.org/en/latest/fixture.html)). If there's something wrong with the initialization of ``SkyModelMapEvaluator`` the tests will fail on creating the fixture. I think the initialization of the class should be part of the test suite.

Reply:
@joleroi - I don't think it's possible to avoid the execution of constructors in fixtures. The main point of using fixtures is to avoid fixture / setup code duplication.
See the first example in the pytest fixture docs, which does the same thing:
https://docs.pytest.org/en/latest/fixture.html

@joleroi - If you think there's a better way to structure the fixture code, or generally want to discuss and improve our current wild mix of test coding patterns in Gammapy, please open a separate issue. (I want this PR to go in today, and we won't resolve the fixture discussion here).

Reply:
> it's possible

Sorry, I wanted to say: it's **not** possible

Reply:
Also: the error message / traceback when there's an exception raised from a constructor or any code in a fixture is I think just as good as an error comming from a test. So in practice I think it's OK to just use fixtures instead of individual initialiser tests, especially given our coding style of having simple very simple `__init__` methods in Gammapy.